Show service properties and icon in vscode extension

xingsy97 opened the pull request:
1. show service properties in resource tree
2. right click service name then click "View Properties" to view detail
3. show correct icon for socket.io service
4. fix a packing issue in `gulpfile.ts`. 

How to use the extension: https://github.com/Azure/azure-webpubsub/pull/685
![image](https://github.com/Azure/azure-webpubsub/assets/87063252/b81f85f7-8ef4-4bcd-9e87-b5bb4c3e8e08)

![image](https://github.com/Azure/azure-webpubsub/assets/87063252/745f0553-3f2c-48f4-b764-8a114add4dd1)

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,90 @@
+            iconPath: new ThemeIcon("location")
+        }));
+        childs.push(createGenericElement({
+            label: localize('serviceProvisioningState', 'Provisioning State'),
```
keep the properties the same as portal, use `Status`?
![image](https://github.com/Azure/azure-webpubsub/assets/668244/0068063f-1bd6-4079-84dd-10d6113f119d)

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,90 @@
+            iconPath: this.getIconPathForProvisoningState(this.webPubSub.provisioningState)
+        }));
+        childs.push(createGenericElement({
+            label: localize('serviceLocalAuth', 'Local Auth'),
```
no need to show? Keep it similar to what portal's overview shows?

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,90 @@
+            iconPath: new ThemeIcon(this.webPubSub.disableLocalAuth ? "error" : "check"),
+        }));
+        childs.push(createGenericElement({
+            label: localize('servicePublicNetworkAccess', 'Public Network Access'),
```
also no need, it sounds a bit advanced property, and vscode extension is mainly to improve development experience

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,90 @@
+        let tlsNodeDesc = "";
+        if (this.webPubSub.sku?.tier !== KnownWebPubSubSkuTier.Free) {
+            if (this.webPubSub.tls) {
+                tlsNodeDesc = this.webPubSub.tls.clientCertEnabled ? "Client Cert Enabled" : "Client Cert Disabled";
```
if this property is shown, we may also need to provide feature to enable the client cert from vscode extension

xingsy97 commented:
@vicancy 
To keep consistent with portal, 
removed: LocalAuth, TLS, PublicNetworkAccess
added: sku and unit count

Now screenshot:
![image](https://github.com/Azure/azure-webpubsub/assets/87063252/027e2026-dcae-4841-ac9e-ad0091b91adc)

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,71 @@
+            iconPath: new ThemeIcon("settings")
+        }));
+        childs.push(createGenericElement({
+            label: localize('serviceUnitCount', 'Unit'),
```
hide it when the sku is Free ?

xingsy97 replied:
hided

vicancy commented on tools/vscode-azurewebpubsub/src/tree/service/ServicePropertiesItem.ts:
```diff
@@ -0,0 +1,71 @@
+            case "Failed": return new ThemeIcon("error"); 
+            case "Canceled": return new ThemeIcon("close"); 
+            case "Unknown": return new ThemeIcon("question"); 
+            default: return new ThemeIcon("error"); 
```
default to `question`?

xingsy97 replied:
updated
